ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -544,7 +544,7 @@ abstract class EventStorageBackendTemplate(
       WHERE
         create_evs.event_sequential_id ${queryStrategy.anyOf(eventSequentialIds)}
         AND NOT EXISTS (  -- check not archived as of snapshot
-          SELECT 1
+          SELECT /*+ INDEX(consuming_evs PARTICIPANT_EVENTS_CONSUMING_EXERCISE_CONTRACT_ID_IDX) NO_INDEX(consuming_evs PARTICIPANT_EVENTS_CONSUMING_EXERCISE_EVENT_SEQUENTIAL_ID) */ 1
```

Comment:
the using of the contract_id index, and not using of the sequential id index should prevent full table scans on the exercise table, and index hash joins between the sequential ID and contract ID indexes for the same table, like:
```
--------------------------------------------------------------------------------------------------------------------------------------
|   0 | SELECT STATEMENT         |                                                           |    10 | 12760 |  1215   (2)| 00:00:01 |
|   1 |  SORT ORDER BY           |                                                           |    10 | 12760 |  1215   (2)| 00:00:01 |
|*  2 |   HASH JOIN ANTI         |                                                           |    10 | 12760 |  1214   (2)| 00:00:01 |
|*  3 |    TABLE ACCESS FULL     | PARTICIPANT_EVENTS_CREATE                                 |  1000 |  1105K|   732   (3)| 00:00:01 |
|*  4 |    VIEW                  | index$_join$_002                                          | 15414 |  2167K|   481   (0)| 00:00:01 |
|*  5 |     HASH JOIN            |                                                           |       |       |            |          |
|*  6 |      INDEX RANGE SCAN    | PARTICIPANT_EVENTS_CONSUMING_EXERCISE_EVENT_SEQUENTIAL_ID | 15414 |  2167K|    37   (0)| 00:00:01 |
|   7 |      INDEX FAST FULL SCAN| PARTICIPANT_EVENTS_CONSUMING_EXERCISE_CONTRACT_ID_IDX     | 15414 |  2167K|   555   (0)| 00:00:01 |
--------------------------------------------------------------------------------------------------------------------------------------
```